Sourcelink seems to break without a remote git repo

Sourcelink seems to break without a remote git repo.  Should try to figure out if there's a way to disable it until a remote has been used.

```
Starting Target: DotnetPack (==> DotnetTest)
dotnet "pack" "/Users/username/Documents/GitHub/someproj/src/someproj/someproj.fsproj" --configuration Release --output /Users/username/Documents/GitHub/someproj/dist /p:PackageVersion=0.1.0 /p:PackageReleaseNotes="Initial release" /p:SourceLinkCreate=true
Microsoft (R) Build Engine version 15.5.179.9764 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for /Users/username/Documents/GitHub/someproj/src/someproj/someproj.fsproj...
  Restore completed in 370.9 ms for /Users/username/Documents/GitHub/someproj/src/someproj/someproj.fsproj.
  fatal: Not a git repository (or any of the parent directories): .git
/Users/username/.nuget/packages/sourcelink.create.commandline/2.7.3/build/SourceLink.Create.CommandLine.targets(21,5): error MSB3073: The command "git rev-parse --show-toplevel" exited with code 128. [/Users/username/Documents/GitHub/someproj/src/someproj/someproj.fsproj]
  fatal: Not a git repository (or any of the parent directories): .git
/Users/username/.nuget/packages/sourcelink.create.commandline/2.7.3/build/SourceLink.Create.CommandLine.targets(21,5): error MSB3073: The command "git rev-parse --show-toplevel" exited with code 128. [/Users/username/Documents/GitHub/someproj/src/someproj/someproj.fsproj]
  fatal: Not a git repository (or any of the parent directories): .git
/Users/username/.nuget/packages/sourcelink.create.commandline/2.7.3/build/SourceLink.Create.CommandLine.targets(21,5): error MSB3073: The command "git rev-parse --show-toplevel" exited with code 128. [/Users/username/Documents/GitHub/someproj/src/someproj/someproj.fsproj]
```

Or without remote

```
/Users/username/.nuget/packages/sourcelink.create.commandline/2.7.3/build/SourceLink.Create.CommandLine.targets(24,5): error MSB3073: The command "git config --get remote.origin.url" exited with code 1. [/Users/username/Documents/GitHub/someproj/src/someproj/someproj.fsproj]
```

--- a/Content/build.fsx
+++ b/Content/build.fsx
@@ -175,6 +175,14 @@ Target "AssemblyInfo" (fun _ ->
 )
 
 Target "DotnetPack" (fun _ ->
+    // https://github.com/ctaggart/SourceLink relies on a some git depedencies.
+    // Checks if there's a remote.origin.url and the last git hash.
+    let remoteOriginOk, remoteMessages, _ = Git.CommandHelper.runGitCommand "." "config --get remote.origin.url"
+    let sha1Ok, sha1Message, _ = Git.CommandHelper.runGitCommand "." "rev-parse HEAD"
+    let canSourceLink =
+        remoteOriginOk && (remoteMessages |> Seq.length > 0)
+        && sha1Ok &&  (sha1Message |> Seq.length > 0)
+
     !! srcGlob
     |> Seq.iter (fun proj ->
         DotNetCli.Pack (fun c ->
@@ -186,7 +194,7 @@ Target "DotnetPack" (fun _ ->
                     [
                         sprintf "/p:PackageVersion=%s" release.NugetVersion
                         sprintf "/p:PackageReleaseNotes=\"%s\"" (String.Join("\n",release.Notes))
-                        "/p:SourceLinkCreate=true"
+                        sprintf "/p:SourceLinkCreate=%b" canSourceLink
                     ]
             })
     )

